### DISCUSSION TITLE
Viikon 22 päivitys

### DISCUSSION THREAD

**Discussion by AnneliO** [KohaSuomi/Koha#598] [Tiedotteet]
Viikon 22 päivitys tehdään tiistai-aamuna 30.5.2023 ja se sisältää seuraavat korjaukset:

### 1.1 Takaaja/taattava-termien korjaus

**Järjestelmäasetus _NoIssuesChargeGuarantees_**

Prevent a patron from checking out if the patron has guarantees owing in total more than EUR in fines.

Käännös nyt
Estä asiakasta lainaamasta, jos hänen takaajalla on maksuja enemmän kuin EUR maksuja.

Korjattu muotoon
Estä asiakasta lainaamasta, jos hänen taattavillaan on maksuja enemmän kuin EUR maksuja.

**Asiakkaan lainaus- ja tiedot-näytöllä**

Huomio-osiossa näkyvä huomautus takaajalla, jos taattavilla on maksuja:

Charges: Patron's guarantees collectively owe xx.

Käännös nyt
Maksut: Asiakkaan takaajilla on maksuja xx.

Korjattu muotoon
Maksut: Asiakkaan taattavilla on yhteensä maksuja xx.

![kuva](https://github.com/KohaSuomi/Koha/assets/33121325/3448f9fb-001d-49ef-9d44-620d1029cc54)

Liittyvä tiketti: [Kaksi käännösvirhettä liittyen takaajiin/taattaviin - Koha #577](https://github.com/KohaSuomi/Koha/issues/577)

### 1.2 Käännösvirhe taustatyön tiedot -sivulla

Taustatyön tiedot -sivulla on uudelleenindeksoinnin epäonnistuessa teksti "/ records have successfully been reindexed. Some errors occurred.". Se on käännetty muotoon "/ tietueen muokkaus onnistui. Joitain virheitä esiintyi."

Käännös on nyt korjattu muotoon  "/ tietuetta indeksoitu onnistuneesti. Joitain virheitä esiintyi."

![kuva](https://github.com/KohaSuomi/Koha/assets/33121325/324a30a0-4188-491e-83fb-35ce72d749c6)

Liittyvä tiketti: [Käännösvirhe taustatyön tiedot -sivulla - Koha-translations #11](https://github.com/KohaSuomi/Koha-translations/issues/11)

## 2. Taustatyöt

Kohan taustatöissä on ollut varsinkin isommissa kimpoissa hitautta versionvaihdon jälkeen. Tämä johtuu siitä, että tietueiden indeksoinnit ovat tulleet mukaan taustatöihin versiossa 22.11. indeksointeja tehdään mm.

* kuvailutietueen ja niteen tallennuksen yhteydessä
* lainatessa ja palautettaessa (esim. saatavuus tai sijaintikirjastotieto muuttuu)

Varsinkin kiireisiin aikoihin niteitä lainataan ja palautetaan paljon ja jokaisesta tapahtumasta tulee työ taustatöihin. Kun töitä on paljon, jonon purkamiseen menee enemmän aikaa kuin aiemmin. Kaikki muut taustatyöt menevät samaan jonoon ja joutuvat odottamaan vuoroaan. Tämä näkyy käyttäjälle ikään kuin hitautena, kun esimerkiksi niteiden poistotyö ei valmistukaan samantien. Valutus voi myös vaikuttaa hitaalta, jos osakohteet joutuvat odottamaan pitkään indeksointivuoroaan. Ennen indeksoitumista osakohteet eivät löydy tiedonhaussa eivätkä näy emon yhteydessä Osakohteet-välilehdellä.

Taustatöihin menevät mm.:

* Niteiden muokkaukset ja poistot eräajoina
* Tietueiden muokkaukset ja poistot eräajoina
* Auktoriteettitietueiden muokkaukset ja poistot eräajoina
* Varausten muokkaukset ja poistot eräajoina
* Elasticsearchin indeksoinnit

Yhteisöstä tuodaan useampia korjauksia, jotka pitäisi auttaa hitauteen. Indeksointeja varten luodaan oma työjononsa, jolloin muut taustatyöt tulee suoritetuksi nopeammalla tahdilla, vaikka indeksointeja olisi paljon. Kun tämä on todettu toimivaksi, otetaan käyttöön vielä töiden rinnakkaiskäsittely eli yhtä jonoa rupeaa käsittelemään useampi prosessi yhtäaikaa, jolloin jono tulee puretuksi nopeammin.

Samalla tuodaan korjaus ongelmaan, jossa epäonnistuneet taustatyöt jäävät jonossa Aloitettu-tilaan eikä niille tule Päättyi-sarakkeeseen ajankohtaa. Työn tilaksikin pitäisi tulla Epäonnistui, mutta näin ei tapahdu. Ongelma ilmenee silloin tällöin, eikä tilannetta saatu aikaiseksi testien aikana. Seuraamme siis tuotannossa, auttoiko korjaus.

Liittyy tiketteihin: 

[Taustatöissä hitautta - Koha #582](https://github.com/KohaSuomi/Koha/issues/582)
[Osa taustatöistä ei valmistu - Koha #349](https://github.com/KohaSuomi/Koha/issues/349)

**Comment by lmstrand** [comment]
Päivitys tehty 29.5. n. klo 22.10.
